### PR TITLE
Add shell commands to the client

### PR DESCRIPTION
**Pull request by ssalinas**
also adds endpoints to get more individual shell command data

### CODE REVIEW

File: SingularityClient/src/main/java/com/hubspot/singularity/client/SingularityClient.java [hunk 1]
```diff
@@ -799,6 +808,24 @@ public SingularityRequestParent updateIncrementalDeployInstanceCount(Singularity
     return getCollection(requestUri, "scheduled task ids", PENDING_TASK_ID_COLLECTION);
   }
 
+  public SingularityTaskShellCommandRequest startShellCommand(String taskId, SingularityShellCommand shellCommand) {
+    final Function<String, String> requestUri = (host) -> String.format(SHELL_COMMAND_FORMAT, getApiBase(host), taskId);
+
+    return post(requestUri, "start shell command", Optional.of(shellCommand), Optional.of(SingularityTaskShellCommandRequest.class)).orNull();
+  }
+
+  public Collection<SingularityTaskShellCommandHistory> getShellCommandHisotry(String taskId) {
```
Comment (kwm4385) [inline SingularityClient/src/main/java/com/hubspot/singularity/client/SingularityClient.java]: `getShellCommandHisotry` -> `getShellCommandHistory`

Comment (ssalinas) [reply SingularityClient/src/main/java/com/hubspot/singularity/client/SingularityClient.java]: thanks! fixed

File: SingularityClient/src/main/java/com/hubspot/singularity/client/SingularityClient.java [hunk 2]
```diff
@@ -799,6 +808,24 @@ public SingularityRequestParent updateIncrementalDeployInstanceCount(Singularity
...
+  public Collection<SingularityTaskShellCommandHistory> getShellCommandHistory(String taskId) {
+    final Function<String, String> requestUri = (host) -> String.format(SHELL_COMMAND_FORMAT, getApiBase(host), taskId);
+
+    return getCollection(requestUri, "get shell command history", SHELL_COMMAND_HISTORY);
+  }
+
+  public Collection<SingularityTaskShellCommandUpdate> getShellCommandUpdates(SingularityTaskShellCommandRequest shellCommandRequest) {
+    final Function<String, String> requestUri = (host) -> String.format(SHELL_COMMAND_UPDATES_FORMAT, getApiBase(host), shellCommandRequest.getTaskId(), shellCommandRequest.getShellCommand().getName(), shellCommandRequest.getTimestamp());
+
+    return getCollection(requestUri, "get shell command history", SHELL_COMMAND_UPDATES);
```
Comment (darcatron) [inline SingularityClient/src/main/java/com/hubspot/singularity/client/SingularityClient.java]: maybe `get shell command update history` instead just so it's differentiated a bit?

Comment (ssalinas) [reply SingularityClient/src/main/java/com/hubspot/singularity/client/SingularityClient.java]: getCollection is the generic implementation to get a collection of objects, it's a private method. The public one is already named `getShellCommandUpdates`

Comment (ssalinas) [reply SingularityClient/src/main/java/com/hubspot/singularity/client/SingularityClient.java]: oh, you meant the string there, :derp: 👍

**Comment by darcatron** [comment]
🚢  🚢  🚢 ⚓️

**Comment by darcatron** [comment]
🚢

**Pull request merged**